.github/workflows/test.yaml

```diff
@@ -1,5 +1,13 @@
 name: Test and Validate
-on: [push]
+on:
+  push:
+    paths-ignore:
+      - '**.md'
+      - '.github/ISSUE_TEMPLATE/**'
+  pull_request:
```

Comment:
We can run on pull requests, also from forks, because there are no secrets required to run this job.

If we were to add tests that do require secrets in the future (e.g. integration or end-to-end tests) we need to figure out a strategy of how we run those, as [secrets are not passed to the runner when a workflow is triggered from a forked repository](https://docs.github.com/en/actions/security-guides/encrypted-secrets#using-encrypted-secrets-in-a-workflow). We can come up with a strategy when we get there.

Reply:
If we were to add tests that require a secret without updating this rule, the job would simply fail on the missing secret.

Reply:
You can't run those tests together with forks. It would make sense to have then two workflow files.

But maybe mocking can avoid that 🤔

Reply:
This is all quite alright yeah. Thanks for the contribution 😄